src/main/java/io/reactivex/schedulers/Schedulers.java

```diff
@@ -45,15 +46,35 @@
     static final Scheduler NEW_THREAD;
 
     static {
-        SINGLE = RxJavaPlugins.initSingleScheduler(new SingleScheduler());
-
-        COMPUTATION = RxJavaPlugins.initComputationScheduler(new ComputationScheduler());
-
-        IO = RxJavaPlugins.initIoScheduler(new IoScheduler());
+        SINGLE = RxJavaPlugins.initSingleScheduler(new Callable<Scheduler>() {
+            @Override
+            public Scheduler call() throws Exception {
+                return new SingleScheduler();
```

Comment:
Just in case call() is called multiple times by the hook, these could actually return an field of an inner class (singleton holder pattern or what's the name):

```java
static final class SingleHolder {
    static final Scheduler DEFAULT = new SingleScheduler();
}

SINGLE = RxJavaPlugins.initiSingleScheduler(new Callable<Scheduler>() {
    @Override
    public Scheduler call() {
        return SingleHolder.DEFAULT;
    }
});
```